addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1940,4 +1941,447 @@ options.registry.SnippetMove = options.Class.extend({
         }
     },
 });
+
+options.registry.InnerChart = options.Class.extend({
+    custom_events: _.extend({}, options.Class.prototype.custom_events, {
+        'get_custom_colors': '_onGetCustomColors',
+
+    }),
+    events: _.extend({}, options.Class.prototype.events, {
+        'click we-button.add_column': '_onAddColumnClick',
+        'click we-button.add_row': '_onAddRowClick',
+        'click we-button.o_we_matrix_remove_col': '_onRemoveColumnClick',
+        'click we-button.o_we_matrix_remove_row': '_onRemoveRowClick',
+        'blur we-matrix input': '_onFocusOutMatrixInput',
+        'focus we-matrix input': '_onFocusMatrixInput',
+
+    }),
+    themeArray: ['alpha', 'beta', 'gamma', 'delta', 'epsilon'],
+    /**
+     * @override
+     */
+    init: function () {
+        this._super.apply(this, arguments);
+        this.$widgetTarget = this.$target.closest('.s_chart');
```

Comment:
Reading your whole PR, I am not sure why you do not simply write the dataset on the `s_chart` element, do not retarget etc, and simply let the widget create the canvas the library needs and use the `s_char` dataset for the options... or is the library using the dataset on the canvas by itself?